pkg/storage/image.go

```diff
@@ -553,9 +554,16 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 		}
 		return nil, err
 	}
+	registry, err := sysregistriesv2.FindRegistry(nil, imageName)
+	if err != nil {
+		return nil, err
+	}
+	if registry != nil && registry.Blocked {
+		return nil, fmt.Errorf("cannot resolve names from registry %q because it's blocked", imageName)
+	}
```

Comment:
No, this calls `FindRegistry` with an unqualified value.

In the `if domain != ""` case, yes, do that (like, a previous version of this PR did).

In the other case, _within_ the `for _, r := range svc.unqualifiedSearchRegistries` loop, make the same check on the `path.Join(r, rem)` value.

Reply:
@vrothberg Any chance you can make these changes and push to Antonio's branch? or open a new PR carrying all the changes? Thanks!

Reply:
@mrunalp, I opened https://github.com/kubernetes-sigs/cri-o/pull/2059. Feel free to close this one here.